pkg/vmprovider/providers/vsphere/vmprovider_vm_utils.go

```diff
@@ -99,21 +97,21 @@ func GetVirtualMachineClass(
 	return vmClass, nil
 }
 
-func GetVMImageStatusAndContentLibraryUUID(
+func GetVMImageAndContentLibraryUUID(
 	vmCtx context.VirtualMachineContext,
-	k8sClient ctrlclient.Client) (*vmopv1.VirtualMachineImageStatus, string, error) {
+	k8sClient ctrlclient.Client) (*vmopv1.VirtualMachineImageSpec, *vmopv1.VirtualMachineImageStatus, string, error) {
 
 	imageName := vmCtx.VM.Spec.ImageName
 	if lib.IsWCPVMImageRegistryEnabled() {
-		vmImageStatus, err := resolveVMImageStatus(vmCtx, k8sClient, imageName)
+		vmImageSpec, vmImageStatus, err := resolveVMImage(vmCtx, k8sClient, imageName)
```

Comment:
Should we just go ahead and return the `vmImage` object instead of returning the two separately now that we use both?

Reply:
I thought about this but 
-We don't need the vmImageSpec for the Update path at all. Just for the Create path. That's the addition of imageSpec to only the createArgs and not updateArgs
- Making resolveVMImage return the entire vmImage can be done but It's not very simple since we support a Cluster scoped and a namespace scoped VMImage resources (two different vmoperator api types) after the ImageRegistryFSS. Both resource share the same spec/status fields so we return just the spec and status internally..
- The Object metadata for this Images isn't need for us atm.  

We can just keep the diff minimal for this change and not have to go down that path.